homeassistant/components/airvisual/__init__.py

```diff
@@ -316,14 +334,29 @@ async def async_unload_entry(hass, config_entry):
         )
     )
     if unload_ok:
-        hass.data[DOMAIN][DATA_COORDINATOR].pop(config_entry.entry_id)
+        if config_entry.data[CONF_INTEGRATION_TYPE] == INTEGRATION_TYPE_GEOGRAPHY:
+            # Re-calculate the update interval period for any remaining consumes of this
+            # API key
+            async_sync_geo_coordinator_update_intervals(
+                hass,
+                config_entry.data[CONF_API_KEY],
+                async_get_cloud_api_update_interval(
```

Comment:
The config entry that is unloaded or removed will still be registered at this point. So the interval calculation will still take it into account, right?

Reply:
I think this problem still remains, since the `num_consumers` is still calculated based on the registered config entries for this api key. Please explain if you think I'm mistaken.

Reply:
Ahh, good call – my mind is soft today! 😆

Reply:
@MartinHjelmare Let me know what you think of bdd1fa6?